openapi.yml

```diff
@@ -0,0 +1,185 @@
+    put:
+      summary: Endpoint to insert or update resources - returns the resource as JSON
+      responses:
+        '200':
+          description: OK - returns resource
+          content:
+            application/json:
+              schema:
+                type: string
+        '404':
+          description: Not found
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+  /resource/{resourceId}/versions:
+    get:
+      summary: Return the resources
+      responses:
+        content:
+          application/json:
+            schema:
+              type: string
+  parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+  /resource/{resourceId}/version/{timestamp}:
+    get:
+      summary: Return the resources
+      responses:
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+        - name: timestamp
+          in: path
+          description: Timestamp (version) of the resource to retrieve
+          required: true
+          schema:
+            type: string
+  /marc/{resourceId}/job/{username}/{timestamp}:
+    get:
+      summary: Return the marc record for the given resource version
+      responses:
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+        - name: username
+          in: path
+          description: Username of the resource creator
+          required: true
+          schema:
+            type: string
+        - name: timestamp
+          in: path
+          description: Timestamp (version) of the resource to retrieve
+          required: true
+          schema:
+            type: string
+  /marc/{resourceId}/version/{username}/{timestamp}:
+    get:
+      summary: Return the marc record for the given resource version
+      responses:
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+        - name: username
+          in: path
+          description: Username of the resource creator
+          required: true
+          schema:
+            type: string
+        - name: timestamp
+          in: path
+          description: Timestamp (version) of the resource to retrieve
+          required: true
+          schema:
+            type: string
+  /marc/{resourceId}:
+    post:
+      summary: <<< TODO: INSERT SOMETHING HERE >>>
+      responses:
```

Comment:
202 with a location header that provides the URL for checking the status of the MARC conversion job.